### PR TITLE
PythonOperator-based airflow DAG factory

### PR DESCRIPTION
**Pull request by mgasner**

### CODE REVIEW

**Comment by schrockn** [comment]
I think the most interesting thing to discuss on this PR is that this diverges significantly from the containerized case because this dynamically constructed versus the codegen approach of the containerized use case. It is certainly nice to not have to manage code generation, but there is definitely a possible state of the world where execution plan generation is quite expensive. Imagine a dag with 1000's of solids where a large sql file is read per solid (this is not theoretical).

**Comment by mgasner** [comment]
Of course this is right -- I wasn't even thinking about that overhead. Let me sketch the scaffolded version of this as well and we can discuss.

File: python_modules/dagster-airflow/dagster_airflow_tests/test_factory.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            assert '__typename' in result['data']['executePlan']
+            assert result['data']['executePlan']['__typename'] == 'ExecutePlanSuccess'
+            result = list(
+                filter(
+                    lambda x: x['outputName'] == 'result',
+                    result['data']['executePlan']['stepEvents'],
+                )
+            )[0]
+            assert json.loads(
+                re.sub(
```
Comment (schrockn) [inline python_modules/dagster-airflow/dagster_airflow_tests/test_factory.py]: oh man this is pretty nasty. at minimum comment worthy

File: python_modules/dagster-airflow/dagster_airflow/compile.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    solid_order = _coalesce_solid_order(execution_plan)
+
+    steps = defaultdict(list)
+
+    for solid_name, solid_steps in itertools.groupby(
+        execution_plan.topological_steps(), lambda x: x.tags['solid']
+    ):
+        steps[solid_name] += list(solid_steps)
+
+    return [(solid_name, steps[solid_name]) for solid_name in solid_order]
```
Comment (schrockn) [inline python_modules/dagster-airflow/dagster_airflow/compile.py]: might be worth documenting that this returns a list of tuples. I would have expected a Dict<str,List<ExecutionStep>>

File: python_modules/dagster-airflow/dagster_airflow/factory.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            run_id=run_id,
+            step_keys=json.dumps(step_keys),
+            pipeline_name=pipeline.name,
+        )
+        return execute_query_from_cli(repository_container, query, variables=None)
+
+    return python_callable
+
+
+def make_airflow_dag(pipeline, env_config=None, dag_id=None, dag_description=None, dag_kwargs=None):
```
Comment (schrockn) [inline python_modules/dagster-airflow/dagster_airflow/factory.py]: this function really is lovely

File: python_modules/dagster-airflow/dagster_airflow/compile.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import itertools
+
+from collections import defaultdict
+
+
+def _coalesce_solid_order(execution_plan):
+    solid_order = [s.tags['solid'] for s in execution_plan.topological_steps()]
```
Comment (natekupp) [inline python_modules/dagster-airflow/dagster_airflow/compile.py]: maybe you only need to pass the `topological_steps()` output here? then you can reuse on L#24 below

File: python_modules/dagster-airflow/dagster_airflow_tests/test_factory.py (hunk 1 above)
Comment (mgasner) [reply python_modules/dagster-airflow/dagster_airflow_tests/test_factory.py]: Added a comment

File: python_modules/dagster-airflow/dagster_airflow/compile.py (hunk 2 above)
Comment (mgasner) [reply python_modules/dagster-airflow/dagster_airflow/compile.py]: great, I'll use an OrderedDict

Comment (schrockn) [reply python_modules/dagster-airflow/dagster_airflow/compile.py]: defaultdict(list) is nice for dealing with this stuff

**Pull request merged**